google_guest_agent/addresses.go

```diff
@@ -446,99 +424,13 @@ func (a *addressMgr) Set(ctx context.Context) error {
 	return nil
 }
 
-// Enables or disables IPv6 on network interfaces.
-func configureIPv6(ctx context.Context) error {
-	var newNi, oldNi metadata.NetworkInterfaces
-	if len(newMetadata.Instance.NetworkInterfaces) == 0 {
-		return fmt.Errorf("no interfaces found in metadata")
-	}
-	newNi = newMetadata.Instance.NetworkInterfaces[0]
-	if len(oldMetadata.Instance.NetworkInterfaces) > 0 {
-		oldNi = oldMetadata.Instance.NetworkInterfaces[0]
-	}
-	iface, err := getInterfaceByMAC(newNi.Mac)
-	if err != nil {
-		return err
-	}
-	switch {
-	case oldNi.DHCPv6Refresh != "" && newNi.DHCPv6Refresh == "",
-		newNi.DHCPv6Refresh == "" && len(oldMetadata.Instance.NetworkInterfaces) == 0:
-		// disable
-		// uses empty old interface slice to indicate this is first-run.
-
-		// Before obtaining or releasing an IPv6 lease, we wait for
-		// 'tentative' IPs as part of SLAAC. We wait up to 5 seconds
-		// for this condition to automatically resolve.
-		tentative := []string{"-6", "-o", "a", "s", "dev", iface.Name, "scope", "link", "tentative"}
-		for i := 0; i < 5; i++ {
-			res := run.WithOutput(ctx, "ip", tentative...)
-			if res.ExitCode == 0 && res.StdOut == "" {
-				break
-			}
-			time.Sleep(1 * time.Second)
-		}
-		if err := run.Quiet(ctx, "dhclient", "-r", "-6", "-1", "-v", iface.Name); err != nil {
-			return err
-		}
-	case oldNi.DHCPv6Refresh == "" && newNi.DHCPv6Refresh != "":
-		// enable
-		tentative := []string{"-6", "-o", "a", "s", "dev", iface.Name, "scope", "link", "tentative"}
-		for i := 0; i < 5; i++ {
-			res := run.WithOutput(ctx, "ip", tentative...)
-			if res.ExitCode == 0 && res.StdOut == "" {
-				break
-			}
-			time.Sleep(1 * time.Second)
-		}
-		val := fmt.Sprintf("net.ipv6.conf.%s.accept_ra_rt_info_max_plen=128", iface.Name)
-		if err := run.Quiet(ctx, "sysctl", val); err != nil {
-			return err
-		}
-		if err := run.Quiet(ctx, "dhclient", "-1", "-6", "-v", iface.Name); err != nil {
-			return err
-		}
-	}
-	return nil
-}
-
-// enableNetworkInterfaces runs `dhclient eth1 eth2 ... ethN`
-// and `dhclient -6 eth1 eth2 ... ethN`.
-// On RHEL7, it also calls disableNM for each interface.
-// On SLES, it calls enableSLESInterfaces instead of dhclient.
+// enableNetworkInterfaces sets up network interfaces using the network manager that is
+// managing the primary network interface.
 func enableNetworkInterfaces(ctx context.Context, config *cfg.Sections) error {
 	if len(newMetadata.Instance.NetworkInterfaces) < 2 {
```

Comment:
I believe this check is preventing us to setup ipv6 for primary interface.

Reply:
Done.